### ISSUE TITLE
training problems

### ISSUE THREAD

**Issue by xiphixiongfei** [msingh9/cs230-Coronary-Calcium-Scoring-#1]
dear msingh9,
      I appreciate your efforts in sharing the codes. However, when try to reproduce your results, I've encountered issues.  I run data_process.py, then run train.py following the example you offered to train models, but the loss and acc are hardly changed since epoch 2, Could you please help me to check what are the problems am I encountered?

**Comment by msingh9** [comment]
Can you experiment with different loss functions? Please try focal loss. It has been a while I looked at this project. I will try to dig some more information and then find the version that worked. My example command may not be the exact one we used during our project.

**Comment by xiphixiongfei** [comment]
I tried different loss, different learning rate, and different upsample_ps, but all the measures above are useless. I also delete the exit() function in data_process.py, so that I can generate the train_dev_pids.dump file, but it has no impact on the issue I encountered. Thanks again for your help!

**Comment by msingh9** [comment]
I spend some time during the weekend, and I can reproduce the problem you are mentioning. I do remember we had same issue during early times. It is hard to train because of small ROI and few few images with positive samples. Please try to train all the images with positive samples and then start to introduce negative samples slowly. Also try attention UNET, (coded up in uneta.py as described in paper https://arxiv.org/pdf/1810.07842.pdf). Uneta shall help.

**Comment by xiphixiongfei** [comment]
Thanks lots for your advice, when I add batch normalization, the model can be trained. But I use only positive samples, focal loss and attention UNET to reproduce your results, there is another issue happened. Although the training set curve is getting better, it doesn't seem to work on the validation set, the curve I get is shown bellow:
![image](https://github.com/msingh9/cs230-Coronary-Calcium-Scoring-/assets/13795307/f217e5b2-a4f3-4f8a-803e-0c8c77d25a67)
Do you know what the problem is? Any help you give is greatly appreciated.
